reproduce/test.py

```diff
@@ -438,10 +528,20 @@ def predict_from_video(opt):
                 else:
                     from_tracker.append(True)
                     bboxes = [last_known_valid_bbox]
-                selected_bbox = select_face(bboxes, frame, face_classifier_model, face_classifier_data_transforms, hor, ver, opt)
+                
+                if opt.use_facerec and last_known_valid_bbox: #Only use facerec if its ready
+                    if opt.facerec == "bbox" and len(fr.known_faces) == 0: #If no known faces, generate a reference image
+                        
+                        fr.generate_ref_image(fr.convert_bounding_boxes([last_known_valid_bbox]), frame)
+                    selected_bbox = fr.select_face(bboxes, frame)
```

Comment:
Is this running every frame?

Reply:
Nope, only run once. Once the reference is generated from the last known bounding box, that known_faces length is updates and this won't be ran again.

Reply:
Ah sorry- correct, same as the original select_face.